src/Symmetric.jl

```diff
@@ -0,0 +1,217 @@
+@doc doc"""
+    SymmetricMatrices{n,T} <: Manifold
+
+The [`Manifold`](@ref) $ \operatorname{Sym} (n)$ consisting of the `T`-valued, symmetric matrices of size $ n\times n$, i.e. the set 
+
+$\operatorname{Sym}(n) = \bigl\{A \in \mathbb F^{n\times n} \big| A^{\mathrm{T}} = A \bigr\}$
+
+where $\mathbb F \in \{ \mathbb R, \mathbb C\}$.
+For `T`, there is a distinction between 
+* `T = Real`: real-valued matrix entries
+* `T = Complex`: complex-valued matrix entries
+
+Being slightly inefficient, the matrices are safed as $n\times n$ arrays despite the redundance of the lower triangle.
+
+
+# Constructor
+
+    SymmetricMatrices(n::Int,T::Type = Real)
+
+"""
+struct SymmetricMatrices{N,T} <: Manifold end
+SymmetricMatrices(N::Int,T::Type = Real) = SymmetricMatrices{N,T}()
+
+function representation_size(::SymmetricMatrices{N,T}) where {N,T}
+    return (N,N)
+end
+
+@doc doc"""
+    manifold_dimension(M::SymmetricMatrices{n,Real})
+
+Return the dimension of the [`SymmetricMatrices`](@ref) matrix `M` with real-valued entries, i.e. 
+````math
+    \frac{n(n+1)}{2}
+````
+"""
+manifold_dimension(M::SymmetricMatrices{N,Real}) where {N,T} = div(N*(N+1),2)
```

Comment:
Here `T` is an unused type parameter

Reply:
Oh, I did not know you can leave that out. Thanks

Reply:
```suggestion
manifold_dimension(M::SymmetricMatrices{N,Real}) where {N} = div(N*(N+1),2)
```
I think you missed a `T` here, still

Reply:
Yes when ending type parameters are left out then they are implicitly added to the end so `::SymmetricMatrices{N}` is `SymmetricMatrices{N,T} where {T}`, which makes dispatch really convenient if one has many type parameters but most commonly dispatches on the first few.